src/lib.js

```diff
@@ -677,6 +676,77 @@ var program,
       }
     },
 
+    /**
+     * Escape JSON strings before JSON.parse
+     *
+     * @param {string} jsonString - Input JSON string
+     * @returns {string} - JSON string with escaped characters
+     */
+    escapeJson: function (jsonString) {
+      // eslint-disable-next-line no-implicit-globals
+      meta = { // table of character substitutions
+        '\t': '\\t',
+        '\n': '\\n',
+        '\f': '\\f',
+        '\r': '\\r'
+      };
+      return jsonString.replace(/[\t\n\f\r]/g, (char) => {
+        return meta[char];
+      });
+    },
+
+    /**
+     * Identifies whether the input data string is a graphql query or not
+     *
+     * @param {string} dataString - Input data string to check if it is a graphql query
+     * @param {string} contentType - Content type header value
+     * @returns {Object} - { result: true, graphql: {Object} } if dataString is a graphql query else { result: false }
+    */
+    identifyGraphqlRequest: function (dataString, contentType) {
+      try {
+        const rawDataObj = _.attempt(JSON.parse, this.escapeJson(dataString));
+        if (contentType === 'application/json' && rawDataObj && !_.isError(rawDataObj)) {
+          if (!_.has(rawDataObj, 'query') || !_.isString(rawDataObj.query)) {
+            return { result: false };
+          }
+          if (_.has(rawDataObj, 'variables')) {
+            if (!_.isObject(rawDataObj.variables)) {
+              return { result: false };
+            }
+          }
+          else {
+            rawDataObj.variables = {};
+          }
+          if (_.has(rawDataObj, 'operationName')) {
+            if (!_.isString(rawDataObj.operationName)) {
+              return { result: false };
+            }
+          }
+          else {
+            rawDataObj.operationName = '';
+          }
+          if (_.keys(rawDataObj).length === 3) {
+            if (_.intersection(_.keys(rawDataObj), ['query', 'variables', 'operationName']).length === 3) {
+              const graphqlVariables = JSON.stringify(rawDataObj.variables, null, 2);
+              return {
+                result: true,
+                graphql: {
+                  query: rawDataObj.query,
+                  operationName: rawDataObj.operationName,
```

Comment:
@aman-v-singh Does postman collection support `operationName` field? If it doesn't we can avoid adding it here.

Reply:
@VShingala Maybe it will be useful for when the module is used independently.

Reply:
@aman-v-singh Hmm, I think it's fine as we anyways only select particular field from this result in App so it shouldn't affect v1 format.